src/withStyles.jsx

```diff
@@ -56,15 +75,7 @@ export function withStyles(
           ThemedStyleSheet.flush();
         }
 
-        const addedProps = {
-          [themePropName]: ThemedStyleSheet.get(themeName),
```

Comment:
yes, it's guaranteed. `get(x) === get(x)`. always. Similarly, `styleDef(x) === styleDef(x)` always, and only does a meaningful amount of work (ie, more than a hash lookup) on the first call. As a result, i'm not really sure that this PR will actually have a meaningful impact on performance, and could even be negative, due to the extra lifecycle events. Have we benchmarked any of this?